cpp/src/parquet/column_writer.cc

```diff
@@ -479,7 +483,11 @@ class SerializedPageWriter : public PageWriter {
         page.repetition_levels_byte_length());
 
     data_page_header.__set_is_compressed(page.is_compressed());
-    data_page_header.__set_statistics(ToThrift(page.statistics()));
+
+    // Write page statistics only when page index is not enabled.
```

Comment:
what if that Page Index is canceled, like all values is nan?

Reply:
In this case, the page stats in the header is also not very useful. Actually in the parquet-mr, it does not write page stats into the header any more.